builtin/credential/cert/backend_test.go

```diff
@@ -1111,15 +1203,32 @@ func TestBackend_organizationalUnit_singleCert(t *testing.T) {
 
 // Test a self-signed client with URI alt names (root CA) that is trusted
 func TestBackend_uri_singleCert(t *testing.T) {
-	connState, err := testConnState(
-		"test-fixtures/root/rootcawuricert.pem",
-		"test-fixtures/root/rootcawurikey.pem",
-		"test-fixtures/root/rootcacert.pem",
-	)
+	u, err := url.Parse("spiffe://example.com/host")
+	if err != nil {
+		t.Fatal(err)
+	}
+	certTemplate := &x509.Certificate{
+		Subject: pkix.Name{
+			CommonName: "example.com",
+		},
+		DNSNames:    []string{"example.com"},
+		IPAddresses: []net.IP{net.ParseIP("127.0.0.1")},
+		URIs:        []*url.URL{u},
+		ExtKeyUsage: []x509.ExtKeyUsage{
+			x509.ExtKeyUsageServerAuth,
+			x509.ExtKeyUsageClientAuth,
+		},
+		KeyUsage:     x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment | x509.KeyUsageKeyAgreement,
+		SerialNumber: big.NewInt(mathrand.Int63()),
+		NotBefore:    time.Now().Add(-30 * time.Second),
+		NotAfter:     time.Now().Add(262980 * time.Hour),
+	}
+
+	tempDir, connState, err := generateTestCertAndConnState(t, certTemplate)
```

Comment:
defer cleanup?

Reply:
Nah, it's a temp dir. This way if the test fails we can access the offending files; when run in CI it gets blown away and when not run in CI it gets blown away eventually anyways.